### PR TITLE
refactor(lang): convert javascript code to typescript code

### PR DESCRIPTION
**Pull request by MustafaAliDS**
This PR converts JavaScript code to TypeScript.

Here is a comparison between the old code and the new code.

JavaScript:

```
const express = require("express");
const querystring = require("querystring");
const request = require("request");
require("dotenv").config();

const app = express();
const PORT = process.env.PORT;
let client_id = process.env.client_id;
let client_secret = process.env.client_secret;

let redirect_uri = `http://localhost:${PORT}/callback`;
let scope = "user-read-private user-read-email";

app.get("/", (req, res) => {
  res.redirect("/login");
});

app.get("/login", (req, res) => {
  res.redirect(
    "https://accounts.spotify.com/authorize?" +
      querystring.stringify({
        response_type: "code",
        client_id: client_id,
        scope: scope,
        redirect_uri: redirect_uri,
      })
  );
});

app.get("/callback", (req, res) => {
  let code = req.query.code || null;

  let authOptions = {
    url: "https://accounts.spotify.com/api/token",
    form: {
      code: code,
      redirect_uri: redirect_uri,
      grant_type: "authorization_code",
    },
    headers: {
      Authorization:
        "Basic " +
        new Buffer(client_id + ":" + client_secret).toString("base64"),
    },
    json: true,
  };

  request.post(authOptions, function (error, response, body) {
    if (!error && response.statusCode === 200) {
      let access_token = body.access_token;
      let token_type = body.token_type;
      let scope = body.scope;
      let expires_in = body.expires_in;

      res.send({
        access_token: access_token,
        token_type: token_type,
        scope: scope,
        expires_in: expires_in,
      });
    }
  });
});

app.listen(PORT, () => {
  console.log(`Working on port ${PORT}`);
});

```

TypeScript:

```
import express, { Application, Request, Response } from "express";
import querystring from "querystring";
import request from "request";
import dotenv from "dotenv";
dotenv.config();

const server: Application = express();
const PORT = 8888;
const CLIENT_ID = "";
const CLIENT_SECERET = "";
const RESPONSE_TYPE = "code";

const REDIRECT_URI = "http://localhost:8888/callback";
const SCOPE =
  "user-read-recently-played user-read-email user-read-private user-read-recently-played user-read-playback-position user-read-playback-state ugc-image-upload";

interface Body {
  access_token: string;
  token_type: string;
  scope: string;
  expires_in: string;
}

server.get("/", (_req: Request, res: Response) => {
  res.redirect("/login");
});

server.get("/login", (req: Request, res: Response) => {
  res.redirect(
    "https://accounts.spotify.com/authorize?" +
      querystring.stringify({
        response_type: RESPONSE_TYPE,
        client_id: CLIENT_ID,
        scope: SCOPE,
        redirect_uri: REDIRECT_URI,
      })
  );
  req.body;
});

server.get("/callback", (req: Request, res: Response) => {
  const COMBINED_IDS = `${CLIENT_ID}:${CLIENT_SECERET}`;

  const AUTH_OPTIONS: {
    url: string;
    form: {
      code: typeof req.query["code"];
      redirect_uri: string;
      grant_type: string;
    };
    headers: {
      Authorization: string;
    };
    json: boolean;
  } = {
    url: "https://accounts.spotify.com/api/token",
    form: {
      code: req.query["code"],
      redirect_uri: REDIRECT_URI,
      grant_type: "authorization_code",
    },
    headers: {
      Authorization: `Basic ${Buffer.from(COMBINED_IDS).toString("base64")}`,
    },
    json: true,
  };

  request.post(AUTH_OPTIONS, function (error, response, body: Body) {
    if (!error && response.statusCode === 200) {
      const ACCESS_TOKEN = body.access_token;
      const TOKEN_TYPE = body.token_type;
      const SCOPE = body.scope;
      const EXPIRES_IN = body.expires_in;

      res.send({
        access_token: ACCESS_TOKEN,
        token_type: TOKEN_TYPE,
        scope: SCOPE,
        expires_in: EXPIRES_IN,
      });
    }
  });
});

server.listen(PORT, () => {
  console.log(`Working on port ${PORT}`);
});


```

Closes https://github.com/MustafaAliDS/SpotifyAPI-Project/issues/1

### CODE REVIEW

**Comment by MustafaAliDS** [comment]
https://prod.liveshare.vsengsaas.visualstudio.com/join?9BCC2770C77E3C48AB1B33B154DBC066FAEF

File: src/server.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import express, { Application, Request, Response } from "express";
+import querystring from "querystring";
+import request from "request";
+import dotenv from "dotenv";
+dotenv.config();
+
+const APP: Application = express();
```
Comment (MichaelAlexanderBanuelos) [inline src/server.ts:7]:
```suggestion
const server: Application = express();
```

`nit`: we should only the all caps pattern when dealing with strings that never change

File: src/server.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+const PORT = process.env["PORT"]
+const CLIENT_ID = process.env["CLIENT_ID"];
+const CLIENT_SECERET = process.env["CLIENT_SECERET"];
```
Comment (MichaelAlexanderBanuelos) [inline src/server.ts:10]: ❤️

File: src/server.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+const REDIRECT_URI = "http://localhost:8888/callback";
+const SCOPE =
+  "user-read-recently-played user-read-email user-read-private user-read-recently-played user-read-playback-position user-read-playback-state ugc-image-upload";
+
+APP.get("/", (req: Request, res: Response) => {
+  res.redirect("/login");
+  console.log(req);
+  console.log(CLIENT_ID);
+  console.log(PORT);
+  console.log(CLIENT_SECERET);
```
Comment (MichaelAlexanderBanuelos) [inline src/server.ts:22]: we should be able to remove these and in general shouldn't have `console.log`s on `main` unless truly needed

File: src/server.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    },
+    json: true,
+  };
+
+  request.post(authOptions, function (error, response, body) {
+    if (!error && response.statusCode === 200) {
+      let access_token: string = body.access_token;
+      let token_type: string = body.token_type;
+      let scope: string = body.scope;
+      let expires_in: string = body.expires_in;
```
Comment (MichaelAlexanderBanuelos) [inline src/server.ts:70]:
```suggestion
      let access_token = body.access_token;
      let token_type = body.token_type;
      let scope = body.scope;
      let expires_in = body.expires_in;
```

`nit`: the typescript compiler should be able to infer the type at this point

**Review by MichaelAlexanderBanuelos — COMMENTED** [review]
this is a great start. i can approve after the comments have been resolved

**Comment by MichaelAlexanderBanuelos** [comment]
another thing we can do that will be needed as we expand this server, is creating a routes file / controller that handles different paths

File: src/server.ts (hunk 4 above)
Comment (MustafaAliDS) [reply src/server.ts:70]: Removing them and hovering over the variables shows "let variablesName: any". Is that fine? 

infer

File: src/server.ts (hunk 3 above)
Comment (MichaelAlexanderBanuelos) [reply src/server.ts:22]: can you try using `_` instead of `req` and see if it still complains about it not being used?

File: src/server.ts (hunk 4 above)
Comment (MichaelAlexanderBanuelos) [reply src/server.ts:70]: _interesting_. lets leave it for now

**Review by MichaelAlexanderBanuelos — APPROVED** [review]
LGTM

**Review by MichaelAlexanderBanuelos — APPROVED** [review]
LGTM

File: src/server.ts [hunk 5]
```diff
@@ -5,15 +5,22 @@ import dotenv from "dotenv";
...
+const PORT = 8888;
+const CLIENT_ID = "";
+const CLIENT_SECERET = "";
 const RESPONSE_TYPE = "code";
 
 const REDIRECT_URI = "http://localhost:8888/callback";
 const SCOPE =
   "user-read-recently-played user-read-email user-read-private user-read-recently-played user-read-playback-position user-read-playback-state ugc-image-upload";
 
+interface body {
```
Comment (MichaelAlexanderBanuelos) [inline src/server.ts:17]: typically we define types and interfaces in [PascalCase](https://stackoverflow.com/a/29603511)

**Review by MichaelAlexanderBanuelos — CHANGES_REQUESTED** [review]

**Review by MichaelAlexanderBanuelos — APPROVED** [review]

**Pull request merged**